web/src/engine/websites/SpoilerPlus.ts

```diff
@@ -0,0 +1,189 @@
+            }).map(xorIt).map(a => {
+                return String.fromCharCode(a);
+            }).join('');
+        }
+
+        const magicString = 'TeH4lL9P3byk27t0';
+        const c = window.__M_CONFIG.c || '';
+        const url = window.location.protocol + '//' + window.location.host;
+        const i = window.__M_CONFIG.i || '';
+        const s = window.__M_CONFIG.s || false;
+
+        const cdn = XOR(url, c);
+        const images = [...document.querySelectorAll('div#post-comic div[data-z]')].map(image => new URL(image.dataset.z, cdn).href);
+
+        if (!s) {
+            resolve({images : images, scrambleArray : undefined });
+        } else {
+            const scrambleSeed = XOR(magicString, i);
+            let difference = 0;
+            const divClasses = document.getElementById('cchapter') ? [...document.getElementById('cchapter').classList]: undefined;
+            const ccChapterRgx = /cchapter-(\\d+)/;
+            if (divClasses) {
+                const goodClass = divClasses.find(element => ccChapterRgx.test(element));
+                if (goodClass) {
+                    difference = parseInt(goodClass.match(ccChapterRgx)[1]);
+                }
+            }
+            const scrambleArray = scrambleSeed.match(/.{1,2}/g).map(a => {
+                return parseInt(a) - difference;
+            });
+            resolve({images : images, scrambleArray : scrambleArray });
+        }
+    });
+
+`;
+
+type PageData = {
+    scrambleArray: number[],
+    images: string[]
+}
+
+type TDimension = {
+    height: number,
+    width: number
+}
+
+type TPiece = {
+    height: number,
+    left: number,
+    top: number,
+    width: number
+}
+
+type TPuzzleData = {
+    from: TPiece,
+    to: TPiece
+}
+
+function MangaLabelExtractor(element: HTMLElement): string {
+    return StripCrap(element.textContent);
+}
+function MangaInfoExtractor(anchor: HTMLAnchorElement) {
+    return {
+        id: anchor.pathname,
+        title: StripCrap(anchor.title)
+    };
+}
+
+@Common.MangaCSS(/^{origin}\/[^/]+-raw-free\/$/, 'article#item-detail h1.title-detail', MangaLabelExtractor)
+@Common.MangasMultiPageCSS('/page/{page}/', 'article.item div.image > a', 1, 1, 0, MangaInfoExtractor)
+@Common.ChaptersSinglePageCSS('div.list-chapter ul li a')
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('spoilerplus', 'SpoilerPlus', 'https://spoilerplus.tv', Tags.Media.Manga, Tags.Language.Japanese, Tags.Source.Aggregator, Tags.Accessibility.RegionLocked);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const data = await FetchWindowScript<PageData>(new Request(new URL(chapter.Identifier, this.URI)), pageScript, 500);
+        return data.images.map(image => new Page(this, chapter, new URL(image), { Referer: this.URI.origin, scrambleArray: JSON.stringify(data.scrambleArray) }));
```

Comment:
Is it really necessary to serialise the scramble array to JSON?

Reply:
Parameters dont accept Arrays.
Its defined as string, number, boolean, ArrayBuffer.

I dont want to change this core definition.